controllers/resources/service.go

```diff
@@ -39,14 +43,38 @@ func NewServiceHandler(client client.Client) ServiceHandler {
 }
 
 func (r *serviceHandler) FetchService(ctx context.Context, log logr.Logger, key types.NamespacedName) (*v1.Service, error) {
-	route := &v1.Service{}
-	err := r.client.Get(ctx, key, route)
+	svc := &v1.Service{}
+	err := r.client.Get(ctx, key, svc)
 	if err != nil && errors.IsNotFound(err) {
 		log.V(1).Info("Service not found.")
 		return nil, nil
 	} else if err != nil {
 		return nil, err
 	}
 	log.V(1).Info("Successfully fetch deployed Service")
-	return route, nil
+	return svc, nil
+}
+
+func (r *serviceHandler) FetchWithRetryAndDelay(ctx context.Context, log logr.Logger, key types.NamespacedName, retryDelay time.Duration, retry int) (*v1.Service, error) {
+	var svc *v1.Service
+	var err error
+
+	for attempt := 1; attempt <= retry; attempt++ {
+		svc, err = r.FetchService(ctx, log, key)
+		if err != nil {
+			if svc == nil || errors.IsNotFound(err) {
+				log.Info(fmt.Sprintf("Service not found. Retrying(%d)..", attempt))
+				time.Sleep(retryDelay)
+				continue
+			} else {
+				return nil, err
+			}
+		}
+		break
+	}
+
+	if svc == nil {
+		log.Info(fmt.Sprintf("Failed to fetch the Service(%s) after retries(%d)", key.Name, retry))
```

Comment:
at this point, the service was not found instead of a failure.

Reply:
Right I changed it to Error and return err